state/linklayerdevices.go

```diff
@@ -525,21 +525,53 @@ func (dev *LinkLayerDevice) RemoveAddresses() error {
 	return dev.st.db().RunTransaction(ops)
 }
 
-// EthernetDeviceForBridge returns LinkLayerDeviceArgs representing an ethernet
+// EthernetDeviceForBridge an InterfaceInfo representing an ethernet
 // device with the input name and this device as its parent.
+// The detail supplied reflects whether the provider is expected to supply the
+// interface's eventual address.
 // If the device is not a bridge, an error is returned.
-func (dev *LinkLayerDevice) EthernetDeviceForBridge(name string) (LinkLayerDeviceArgs, error) {
+func (dev *LinkLayerDevice) EthernetDeviceForBridge(
+	name string, askProviderForAddress bool,
+) (network.InterfaceInfo, error) {
+	var newDev network.InterfaceInfo
+
 	if dev.Type() != network.BridgeDevice {
-		return LinkLayerDeviceArgs{}, errors.Errorf("device must be a Bridge Device, receiver has type %q", dev.Type())
+		return newDev, errors.Errorf("device must be a Bridge Device, but is type %q", dev.Type())
 	}
 
-	return LinkLayerDeviceArgs{
-		Name:        name,
-		Type:        network.EthernetDevice,
-		MACAddress:  network.GenerateVirtualMACAddress(),
-		MTU:         dev.MTU(),
-		IsUp:        true,
-		IsAutoStart: true,
-		ParentName:  dev.ID(),
-	}, nil
+	addrs, err := dev.Addresses()
+	if err != nil {
+		return network.InterfaceInfo{}, errors.Trace(err)
+	}
+
+	newDev = network.InterfaceInfo{
+		InterfaceName:       name,
+		MACAddress:          network.GenerateVirtualMACAddress(),
+		ConfigType:          network.ConfigManual,
+		InterfaceType:       network.EthernetInterface,
+		MTU:                 int(dev.MTU()),
+		ParentInterfaceName: dev.Name(),
+	}
+
+	if len(addrs) > 0 {
```

Comment:
Should we error if no addresses can be established? Otherwise, the CIDR field will be left blank

Reply:
This replicates the prior logic. Possibly, but I don't want to go further down the rabbit hole at this point.